src/py_sonar_scanner/environment.py

```diff
@@ -17,67 +17,90 @@
 
-    def _install_scanner(self, system_name: str):
-        os.mkdir(self.cfg.sonar_scanner_path)
-        # Download the binaries and unzip them
-        # https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-${version}-${os}.zip
-        scanner_res = urllib.request.urlopen(
-            f'https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-{self.cfg.sonar_scanner_version}-{system_name}.zip')
-        scanner_zip_path = os.path.join(self.cfg.sonar_scanner_path, 'scanner.zip')
-        with open(scanner_zip_path, 'wb') as output:
-            output.write(scanner_res.read())
-        with zipfile.ZipFile(scanner_zip_path, "r") as zip_ref:
-            zip_ref.extractall(self.cfg.sonar_scanner_path)
-        os.remove(scanner_zip_path)
-        self._change_permissions_recursive(self.cfg.sonar_scanner_path, 0o777)
+    def scanner(self) -> Scanner:
+        return Scanner(self.cfg)
 
     def _is_sonar_scanner_on_path(self) -> bool:
-        return shutil.which('sonar-scanner') is not None
+        return shutil.which("sonar-scanner") is not None
 
     def cleanup(self):
         if os.path.exists(self.cfg.sonar_scanner_path):
             shutil.rmtree(self.cfg.sonar_scanner_path)
 
+    def _install_scanner(self, system_name: str):
+        os.mkdir(self.cfg.sonar_scanner_path)
+        # Download the binaries and unzip them
+        scanner_zip_path = self._download_scanner_binaries(
+            self.cfg.sonar_scanner_path, self.cfg.sonar_scanner_version, system_name
+        )
+        self._unzip_binaries(scanner_zip_path, self.cfg.sonar_scanner_path)
+        os.remove(scanner_zip_path)
+        self._change_permissions_recursive(self.cfg.sonar_scanner_path, 0o777)
+
     def _change_permissions_recursive(self, path, mode):
         for root, dirs, files in os.walk(path, topdown=False):
             for dir in [os.path.join(root, d) for d in dirs]:
                 os.chmod(dir, mode)
             for file in [os.path.join(root, f) for f in files]:
                 os.chmod(file, mode)
 
-    def scanner(self) -> Scanner:
-        return Scanner(self.cfg)
+    def _download_scanner_binaries(self, destination: str, scanner_version: str, system_name: str) -> str:
+        try:
+            scanner_res = urllib.request.urlopen(f"{self.scanner_base_url}-{scanner_version}-{system_name}.zip")
+            scanner_zip_path = os.path.join(destination, "scanner.zip")
+            self._write_binaries(scanner_res, scanner_zip_path)
+            return scanner_zip_path
+        except HTTPError as error:
+            self.log.error(f"ERROR: could not download scanner binaries - {error.code} - {error.msg}")
+            raise error
+
+    def _write_binaries(self, scanner_res: bytes, destination: str):
+        with open(destination, "wb") as output:
+            output.write(scanner_res.read())
+
+    def _unzip_binaries(self, scanner_zip_path: str, destination: str):
```

Comment:
I moved the last two methods to a utils file. But not the _download_scanner_binraies as it is still too strongly linked to the environment. I created [a ticket](https://sonarsource.atlassian.net/browse/PYSCAN-30) so that we do not forget.